src/shared/Microsoft.Git.CredentialManager/IniFileValueStore.cs

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+using System;
+using System.IO;
+
+namespace Microsoft.Git.CredentialManager
+{
+    /// <summary>
+    /// Value store that uses an INI file as the persistent storage.
+    /// </summary>
+    public class IniFileValueStore : ITransactionalValueStore<string, string>
+    {
+        private readonly IFileSystem _fileSystem;
+        private readonly IniSerializer _serializer;
+        private readonly string _filePath;
+        private readonly string _parentPath;
+        private readonly object _fileLock = new object();
+        private IniFile _iniFile;
+
+        public IniFileValueStore(IFileSystem fileSystem, IniSerializer serializer, string filePath)
+        {
+            _fileSystem = fileSystem;
+            _serializer = serializer;
+            _filePath = filePath;
+            _parentPath = Path.GetDirectoryName(_filePath);
+
+            Reload();
+        }
+
+        public void Reload()
+        {
+            lock (_fileLock)
+            {
+                if (_fileSystem.FileExists(_filePath))
+                {
+                    string text = _fileSystem.ReadAllText(_filePath);
+                    using (var reader = new StringReader(text))
+                    {
+                        _iniFile = _serializer.Deserialize(reader);
+                    }
+                }
+                else
+                {
+                    // Create a new empty INI file object
+                    _iniFile = new IniFile();
+                }
+            }
+        }
+
+        public void Commit()
+        {
+            lock (_fileLock)
+            {
+                // Sure parent directory exists
+                if (!_fileSystem.DirectoryExists(_parentPath))
+                {
+                    _fileSystem.CreateDirectory(_parentPath);
+                }
+
+                using (var writer = new StringWriter())
+                {
+                    _serializer.Serialize(_iniFile, writer);
+                    _fileSystem.WriteAllText(_filePath, writer.ToString());
+                }
+            }
+        }
+
+        public bool TryGetValue(string key, out string value)
+        {
+            lock (_fileLock)
```

Comment:
What are we locking here? The name indicates the file, but I believe the data has been loaded at this point.